components/public-api-server/pkg/oidc/router.go

```diff
@@ -122,21 +129,34 @@ func (s *Service) getCallbackHandler() http.HandlerFunc {
 		log.WithField("id_token", result.IDToken).Trace("user verification was successful")
 
 		if state.Activate {
-			err = s.ActivateClientConfig(r.Context(), config)
+			err = s.ActivateAndVerifyClientConfig(r.Context(), config)
 			if err != nil {
 				log.Warn("Failed to mark config as active: " + err.Error())
 				http.Error(rw, "Failed to mark config as active", http.StatusInternalServerError)
 				return
 			}
 		}
 
-		cookie, _, err := s.CreateSession(r.Context(), result, config)
-		if err != nil {
-			log.Warn("Failed to create session: " + err.Error())
-			http.Error(rw, "Failed to create session", http.StatusInternalServerError)
-			return
+		if state.Verify {
```

Comment:
I think we could skip this if `state.Activate` is true. It probably doesn't hurt to mark it verified again, but could avoid the call if we want.